spec/controllers/queue_items_controller_spec.rb

```diff
@@ -66,35 +65,48 @@
     end
     
     context "when a user is unauthenticated" do
-      it "redirects to the sign in page for unauthenticated users" do
-        post :create, video_id: 3
-        expect(response).to redirect_to sign_in_path
+      it_behaves_like "requires sign in" do
+        let(:action) { post :create, video_id: 3 }
       end
     end
   end
 
   describe "DELETE destroy" do
     context "when deleting the queue for the associated user" do
-      let(:user) { Fabricate(:user) }
-      let(:queue_item) { Fabricate(:queue_item, user: user) }
+      before do
+        set_current_user
+      end
+      let(:queue_item) { Fabricate(:queue_item, user_id: User.find(session[:user_id]).id) }
```

Comment:
`User.find(session[:user_id])` you can wrap this to `current_user` method and put it macros.